File: WebClient/src/Message.ts
import { IMessage } from "./IMessage";

export class Message implements IMessage {

    private correlationId: string;
    private content: string;
    private sender: string;
    private topic: string;
    private timeStamp: Date;



    constructor(id: string, content: string, sender: string, topic: string, timeStamp?: Date) {

        this.correlationId = id;
        this.content = content;
        this.sender = sender;
        this.topic = topic;
        this.timeStamp = timeStamp || new Date();
    }

    public get Topic() { return this.topic; }
    public set Topic(value: string) { this.topic = value; }

    get CorrelationId() { return this.correlationId; }
    set CorrelationId(value: string) { this.correlationId = value; }

    get Sender() { return this.sender; }
    set Sender(value: string) { this.sender = value; }

    get Content() { return this.content; }
    set Content(value: string) { this.content = value; }

    get TimeStamp() { return this.timeStamp; }
    set TimeStamp(value: Date) { this.timeStamp = value; }

}

File: Shared/Message.cs
using System;
using Newtonsoft.Json;

namespace SignalRCommunicator
{
    [Serializable]
    public class Message : IMessage
    {
        [JsonProperty]
        public string CorrelationId { get; set; }

        [JsonProperty]
        public string Topic { get; set; }

        [JsonProperty]
        public string Content { get; set; }

        [JsonProperty]
        public string Sender { get; set; }

        [JsonProperty]
        public DateTime Timestamp { get; set; }

        public Message ()
        {
            CorrelationId = Guid.NewGuid().ToString();
        }
        public Message( string topic, string content )
        {
            Topic = topic;
            Content = content;
        }
        public Message( 
            string topic,
            string content,
            string sender) : this()
        {
            Topic = topic;
            Content = content;
            Timestamp = DateTime.Now;
            Sender = sender;
        }

    }
}
